haystack/document_stores/search_engine.py

```diff
@@ -106,6 +106,8 @@ def __init__(
             raise DocumentStoreError(
                 f"Invalid value {similarity} for similarity, choose between 'cosine', 'l2' and 'dot_product'"
             )
+        client_info = self.client.info()
+        self.server_version = tuple(int(num) for num in client_info["version"]["number"].split("."))
```

Comment:
Nice touch, tuple to make it immutable, eh?

Reply:
Yes, also to make it consistent with the type of [`VERSION` from the elasticsearch client](https://github.com/deepset-ai/haystack/blob/63fd63ff238e6b7d705de8190da79d56db71fd40/haystack/document_stores/elasticsearch/__init__.py#L3).